src/anemoi/training/losses/weightedloss.py

```diff
@@ -77,16 +90,17 @@ def scale_by_variable_scaling(
             Scaled error tensor
         """
         # Use feature_weights if available
-        if self.variable_scaling is None:
+        if len(self.scalar) == 0:
             return x
 
-        if feature_indices is None:
-            return x * self.variable_scaling
-        return x * self.variable_scaling[..., feature_indices]
+        scalar = self.scalar.get_scalar(x.ndim)
+
+        if feature_indices is None or "variable_scaling" not in self.scalar:
+            return x * scalar
+        return x * scalar[..., feature_indices]
```

Comment:
Shouldn't the piece of code be in the ScaleTensor class ?

Reply:
Wdym? The scalar indexing? I would say it depends on the `feature_indices`, so it belongs in the loss function.